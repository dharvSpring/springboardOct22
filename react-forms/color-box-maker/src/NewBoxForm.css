
input, label {
    padding: 0.5rem;
    border-radius: 1rem;
    font-size: large;
}