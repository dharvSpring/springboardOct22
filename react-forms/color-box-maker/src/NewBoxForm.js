import React, { useState } from "react";
import { v4 as uuid } from 'uuid';
import './NewBoxForm.css';

function NewBoxForm({callOnSubmit}) {
    const initialState = {
        width: '',
        height: '',
        backgroundColor: '',
    }
    let [formData, setFormData] = useState(initialState);

    const handleChange = (evt) => {
        const {name, value} = evt.target
        setFormData(fData => ({
            ...fData,
            [name]: value,
        }));
    }

    const handleSubmit = (evt) => {
        evt.preventDefault();
        callOnSubmit({
            ...formData,
            id: uuid(),
        });
        setFormData(initialState);
    }

    return (
        <form className="NewBoxForm" onSubmit={handleSubmit}>
            <label htmlFor="backgroundColor">Background Color</label>
            <input
                type="text"
                id="backgroundColor"
                name="backgroundColor"
                placeholder="Background Color"
                value={formData.backgroundColor}
                onChange={handleChange}
            />

            <label htmlFor="width">Width</label>
            <input
                type="text"
                id="width"
                name="width"
                placeholder="Width"
                value={formData.width}
                onChange={handleChange}
            />

<           label htmlFor="height">Height</label>
            <input
                type="text"
                id="height"
                name="height"
                placeholder="Height"
                value={formData.height}
                onChange={handleChange}
            />

            <input type="submit" value="Add Box!" />
        </form>
    )
}

export default NewBoxForm;