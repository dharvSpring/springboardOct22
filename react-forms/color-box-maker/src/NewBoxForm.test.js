import React from "react";
import { render } from "@testing-library/react";

import NewBoxForm from "./NewBoxForm";

it("smoketest", function() {
    render(<NewBoxForm />);
});

it("snapshot", function() {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
});
