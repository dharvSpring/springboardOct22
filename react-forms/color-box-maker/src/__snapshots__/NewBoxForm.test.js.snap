// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewBoxForm"
  >
    <label
      for="backgroundColor"
    >
      Background Color
    </label>
    <input
      id="backgroundColor"
      name="backgroundColor"
      placeholder="Background Color"
      type="text"
      value=""
    />
    <label
      for="width"
    >
      Width
    </label>
    <input
      id="width"
      name="width"
      placeholder="Width"
      type="text"
      value=""
    />
    <label
      for="height"
    >
      Height
    </label>
    <input
      id="height"
      name="height"
      placeholder="Height"
      type="text"
      value=""
    />
    <input
      type="submit"
      value="Add Box!"
    />
  </form>
</DocumentFragment>
`;
